### PR TITLE
fix: add ignore_empty_and_null parameter to `.explode()`

### PR DESCRIPTION
**Pull request by singularityDLW**
Add an optional `ignore_empty` parameter to `explode()` that allows users to drop rows with empty lists or null values instead of producing null rows.

- Default behavior (ignore_empty=False) is unchanged: empty lists and nulls produce a single null row
- When ignore_empty=True: empty lists and nulls are dropped entirely, matching PyArrow/DuckDB behavior

## Changes Made

- Added ignore_empty parameter to Python API (DataFrame.explode, Expression.explode, daft.functions.explode)
- Propagated the parameter through logical plan, physical plan, and execution layers
- Updated Rust core implementation in daft-functions-list and daft-recordbatch
- Added 11 test cases covering basic usage and edge cases (all empty lists, all nulls, FixedSizeList, nested lists, etc.)

## Related Issues

#5752

### CODE REVIEW

File: daft/dataframe/dataframe.py [hunk 1]
```diff
@@ -3047,15 +3047,19 @@ def drop_null(self, *cols: ColumnInputType) -> "DataFrame":
         return self.where(~reduce(lambda x, y: x | y, (x.is_null() for x in columns)))
 
     @DataframePublicAPI
-    def explode(self, *columns: ColumnInputType, index_column: ColumnInputType | None = None) -> "DataFrame":
+    def explode(
+        self, *columns: ColumnInputType, ignore_empty: bool = False, index_column: ColumnInputType | None = None
```
Comment (rchowell) [inline daft/dataframe/dataframe.py]: Please put the new parameter at the end of the list, this way it is not a breaking change for those who pass positional arguments. Thank you.

Comment (singularityDLW) [reply daft/dataframe/dataframe.py]: Hi @rchowell 

I have finished. Please review it. Thanks so much.

**Comment by singularityDLW** [comment]
Hi @srilman and @rchowell,

Could you please take a look at this PR when you have a moment? I'm happy to address any feedback. Thanks for your time!

File: tests/dataframe/test_explode_ignore_empty.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
```
Comment (srilman) [inline tests/dataframe/test_explode_ignore_empty.py]: Rather than creating a completely new file, can we just put this in `tests/dataframe/test_explode.py`. Plus, I think a lot of these tests cover the same cases, so maybe just a handful are enough?

File: daft/dataframe/dataframe.py [hunk 3]
```diff
@@ -3047,16 +3047,20 @@ def drop_null(self, *cols: ColumnInputType) -> "DataFrame":
         return self.where(~reduce(lambda x, y: x | y, (x.is_null() for x in columns)))
 
     @DataframePublicAPI
-    def explode(self, *columns: ColumnInputType, index_column: ColumnInputType | None = None) -> "DataFrame":
+    def explode(
+        self, *columns: ColumnInputType, index_column: ColumnInputType | None = None, ignore_empty: bool = False
```
Comment (srilman) [inline daft/dataframe/dataframe.py]: Actually rather than naming it `ignore_empty`, can we name it `ignore_empty_and_null` to clarify that both cases differ?

File: daft/functions/list.py [hunk 4]
```diff
@@ -348,6 +348,7 @@ def explode(list_expr: Expression) -> Expression:
 
     Args:
         list_expr (List Expression): expression to explode.
+        ignore_empty: whether to drop empty lists or None values. Defaults to False.
```
Comment (srilman) [inline daft/functions/list.py]: Can we add a bit more detail describing the default which will add a single NULL value given an empty or NULL row?

**Review by srilman — APPROVED** [review]
Overall LGTM, just a couple of nits. Thanks @singularityDLW! Let me know when you're ready to merge this PR and I'll do it.

**Comment by singularityDLW** [comment]
Hi @srilman,

I've addressed all the feedback you mentioned:
  - Renamed ignore_empty to ignore_empty_and_null for clarity
  - Improved the docstrings to describe the default behavior
  - Merged tests into test_explode.py and removed redundant test cases

Please review when you have a chance. Let me know if there's anything else that needs to be adjusted. Thanks!
